src/Symfony/Component/Console/Application.php

```diff
@@ -864,14 +864,14 @@ protected function configureIO(InputInterface $input, OutputInterface $output)
 
         if (true === $input->hasParameterOption(['--no-interaction', '-n'], true)) {
             $input->setInteractive(false);
-        } elseif (\function_exists('posix_isatty')) {
+        } else {
             $inputStream = null;
 
             if ($input instanceof StreamableInputInterface) {
                 $inputStream = $input->getStream();
             }
 
-            if (!@posix_isatty($inputStream) && false === getenv('SHELL_INTERACTIVE')) {
+            if (!stream_isatty($inputStream ?: STDIN) && false === getenv('SHELL_INTERACTIVE')) {
```

Comment:
Not sure about this change. Better do this IMHO:
```php
if (!($input instanceof StreamableInputInterface && !stream_isatty($input->getStream()) && false === getenv('SHELL_INTERACTIVE'))) {
```

Reply:
I can't do that, because nothing usually sets the stream, it's NULL most of the time, so this branch would never be triggered. It seems `posix_isatty` uses STDIN internally when argument is `null`, but this is not the case with `stream_isatty`. Same logic is actually in QuestionHelper.

Reply:
ok, then we should make this more robust: `STDIN` is defined only when using the `cli` SAPI, so we should also check its existence before using it.

Reply:
Done